src/genie/libs/parser/asa/show_access_list.py

```diff
@@ -0,0 +1,187 @@
+        # access-list acl4; 5 elements; name hash: 0xffffffff
+        # access-list acl5; 5 elements; name hash: 0xffffffff
+
+        p1 = re.compile(r'^access-list (?P<name>\S+); +(?P<elements>\d+) +elements; +name +hash\: +(?P<namehash>0x[0-9a-fA-F]+)')
+
+        # access-list acl1 line 1 extended permit udp host 10.38.193.26 host 9.0.128.50 eq domain log informational interval 300 (hitcnt=0) 0xffffffff
+        # access-list acl1 line 2 extended permit tcp object-group group-In object-group group-Out eq bgp log disable (hitcnt=0) 0xffffffff 
+        # access-list acl1 line 3 extended deny ip any any log informational interval 300 (hitcnt=60) 0xffffffff
+        # access-list acl1 line 4 extended deny ip any4 any4 log informational interval 300 (hitcnt=0) 0xffffffff
+        # access-list acl1 line 5 extended deny ip any6 any6 log informational interval 300 (hitcnt=0) 0xffffffff
+        p2 = re.compile(r' *access-list +(?P<name>\S+) +line +(?P<entry>\d+) +extended +(?P<action>(permit|deny)) +(?P<protocol>\S+) +(host (?P<src_host>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})|object-group +(?P<src_object_group>\S+)|(?P<dst_any>any|any4|any6))( +eq +(?P<src_port>\S+))? +(host (?P<dst_host>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})|object-group +(?P<dst_object_group>\S+)|(?P<src_any>any|any4|any6))( +eq +(?P<dst_port>\S+))?( +(?P<log>log))?( +(?P<disable>disable))?( +informational +interval +(?P<informational_interval>\d+))? +\(hitcnt=(?P<hitcnt>\d+)\) +(?P<acl_hash>0x[0-9a-fA-F]+)')
+
+        # access-list acl1 line 6 remark this is a remark
+        p3 = re.compile(r' *access-list +(?P<name>\S+) +line +(?P<entry>\d+) +remark+(?P<remark>.*)')
+
+        # access-list cached ACL log flows: total 0, denied 0 (deny-flow-max 4096)
+        p4 = re.compile(r' *access-list +cached +ACL +log +flows: +total +(?P<total>\d+), +denied +(?P<denied>\d+) +\(deny-flow-max +(?P<deny_flow_max>\d+)\)')
+        # alert-interval 300
+        p5 = re.compile(r' *alert-interval +(?P<alert_interval>\d+)')
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            if line == '':
+                continue
+
+            m = p1.match(line)
+            if m:
+                groups = m.groupdict()
+                acl_name = groups['name']
+                ret_dict['access-list'][acl_name] = {}
+                ret_dict['access-list'][acl_name]['elements'] = int(groups['elements'])
+#                ret_dict['access-list'][acl_name]['name_hash'] = groups['namehash']
+                continue
+            
+            m = p2.match(line)
+            if m:
+                groups = m.groupdict()
+                acl_name = groups['name']
+                line_num = groups['entry']
+                if not acl_name in ret_dict['access-list'].keys() :
+                    ret_dict['access-list'][acl_name] = {}
+                if not 'entry' in ret_dict['access-list'][acl_name].keys():
+                    ret_dict['access-list'][acl_name]['entry'] = {}
```

Comment:
The magic of `<dict>.setdefault()` is it wont overwrite existing keys. It will only insert if the key doesnt exist.
``` python
acl_entry_dict = ret_dict.setdefault('access-list', {}).setdefault(acl_name, {}).setdefault('entry', {})